android/modules/geolocation/src/java/ti/modules/titanium/geolocation/TiLocation.java

```diff
@@ -106,7 +106,12 @@ public boolean getLocationServicesEnabled()
 			}
 		}
 
-		return providerNames.size() != 0 && getLastKnownLocation() != null;
+		for (String providerName : providerNames) {
```

Comment:
We want to return true if any location provider is enabled, `GPS`, `NETWORK` etc..

```Java
return providerNames.size() > 0;
```

Reply:
Gary is right. We need to check all providers. His code change is the correct solution.

Also, we should check if the `providerNames` collection is null after calling the `LocationManager.getProviders()` method. If null, return false. (Odds are this isn't going to happen, but better safe than sorry. Especially on Android.)